code/game/atoms.dm

```diff
@@ -93,6 +93,10 @@
 	LAZYCLEARLIST(priority_overlays)
 	//SSoverlays.processing -= src	//we COULD do this, but it's better to just let it fall out of the processing queue
 
+	//anything not dead by this point needs to fucking not exist
+	for(var/I in src)
+		qdel(I)

```

Comment:
after this loop is finished you need to do `contents.Cut()` so that the references to these atoms get removed